spec/unit/serializer_spec.rb

```diff
@@ -0,0 +1,214 @@
+    let(:valid_values)  {  ["Asia/Tokyo", "Asia/Shanghai", "Europe/Berlin"] }
+    let(:parsed_values) {  [TZInfo::Timezone.get("Asia/Tokyo"), TZInfo::Timezone.get("Asia/Shanghai"), TZInfo::Timezone.get("Europe/Berlin")] }
+
+    let(:invalid_values) { ["Tokyo", {}, Object.new, "Eastern"] }
+
+    let(:string_serializer) { :identifier }
+
+    it_behaves_like "a serializer"
+    it_behaves_like "a non-JSON value serializer"
+  end
+
+  describe IknowParams::Serializer::UUID do
+    let(:valid_values)  {  ["e56986f4-4418-4f7a-99d8-3faccde33f6b", "1e851f60-4eb9-4660-ad4e-ac0b315f3647", "8a540cb9-8107-45e8-8611-82b9171b4a4b"] }
+    let(:parsed_values)  {  ["e56986f4-4418-4f7a-99d8-3faccde33f6b", "1e851f60-4eb9-4660-ad4e-ac0b315f3647", "8a540cb9-8107-45e8-8611-82b9171b4a4b"] }
+
+    let(:invalid_values) { ["e56986f4-4418-4f7a-99d8-3faccde33f$b", {}, Object.new, "e56986f4-4418-4f7a-99d8-3faccde33fb"] }
+
+    it_behaves_like "a serializer"
+    it_behaves_like "a JSON-value serializer"
+  end
+
+  # Test abstract JsonWithSchema serializer by making a real schema
+  # and serializer for it.
+
+  class MyJsonSerializer < IknowParams::Serializer::JsonWithSchema
+    def initialize
+      schema = {"type" => "object", "required" => ["a"], "properties" => {"a" => {"type" => "integer"}}}
+      super(schema)
+    end
+
+    set_singleton!
+  end
+
+  describe MyJsonSerializer do
+    let(:valid_values)  { ['{"a": 5}', '{"a": 6}'] }
+    let(:parsed_values)  { [{"a" => 5}, {"a" => 6}] }
+
+    let(:invalid_values) { ['{"a": "5"}', '{"b": 6}', '[]', '{}'] }
+
+    let(:string_serializer) { :to_json }
+
+    it_behaves_like "a serializer"
+    it_behaves_like "a JSON-value serializer"
+  end
+
+  # Test abstract JsonWithSchema serializer by making a real schema
+  # and serializer for it.
+
+  class MyJsonSerializer < IknowParams::Serializer::JsonWithSchema
+    def initialize
+      schema = {"type" => "object", "required" => ["a"], "properties" => {"a" => {"type" => "integer"}}}
+      super(schema)
+    end
+
+    set_singleton!
+  end
+
+  describe MyJsonSerializer do
+    let(:valid_values)  { ['{"a": 5}', '{"a": 6}'] }
+    let(:parsed_values)  { [{"a" => 5}, {"a" => 6}] }
+
+    let(:invalid_values) { ['{"a": "5"}', '{"b": 6}', '[]', '{}'] }
+
+    let(:string_serializer) { :to_json }
+
+    it_behaves_like "a serializer"
+    it_behaves_like "a JSON-value serializer"
+  end
+
+  # Test abstract StringEnum serializer by making a real enum
+  # and serializer for it.
+
+  class MyStringEnumSerializer < IknowParams::Serializer::StringEnum
+    def initialize
+      super("a", "b")
+    end
+
+    set_singleton!
+  end
+
+  describe MyStringEnumSerializer do
+    let(:valid_values)   { ['a', 'b', 'B'] }
+    let(:parsed_values)  { ['a', 'b', 'b'] }
+
+    let(:invalid_values) { ['c', 'd'] }
+
+    let(:string_serializer) { :downcase }
+
+    it_behaves_like "a serializer"
+    it_behaves_like "a non-JSON value serializer"
+  end
+end
+
```

Comment:
You mentioned making PersistentEnum a test dependency; did that prove to require too much scaffolding to write the tests?